controllers/devworkspace/solver/che_routing.go

```diff
@@ -162,13 +163,23 @@ func (c *CheRoutingSolver) provisionPodAdditions(objs *solvers.RoutingObjects, c
 	objs.PodAdditions.Containers = append(objs.PodAdditions.Containers, corev1.Container{
 		Name:            wsGatewayName,
 		Image:           cheCluster.Spec.Gateway.Image,
-		ImagePullPolicy: corev1.PullAlways,
+		ImagePullPolicy: corev1.PullIfNotPresent,
```

Comment:
why do we need this change?

Reply:
to avoid image pull on each and every WS startup

Reply:
We can use `deploy.DefaultPullPolicyFromDockerImage()` to detect Image pull policy.